docs/src/Cli/GenerateCommand.php

```diff
@@ -32,94 +34,96 @@ final class GenerateCommand {
 		$source_folder      = realpath( $source_folder );
 		$destination_folder = realpath( $destination_folder );
 
-		$output_file = $destination_folder . '/docs.json';
-
 		$data = $this->get_phpdoc_data( $source_folder );
 
-		$result = false;
-		switch ( $format ) {
-			case 'json':
-				$json   = wp_json_encode( $data, JSON_PRETTY_PRINT );
-				$result = file_put_contents( $output_file, $json ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
-				break;
-			case 'markdown':
-				try {
-					$doc_tree = new Root( $data );
-				} catch ( Exception $exception ) {
-					WP_CLI::error(
-						"Failed to build documentation object tree: {$exception->getMessage()}\n{$exception->getTraceAsString()}",
-						false // Using separate exit for PHPStan.
-					);
-					exit;
-				}
-
-				$template_engine = new MustacheTemplateEngine();
-
-				try {
-					foreach (
-						$this->generate_markdown( $doc_tree, $template_engine ) as $markdown
-					) {
-						/** @var Markdown $markdown */
-						$filepath = "{$destination_folder}/{$markdown->get_filename()}";
-						$this->ensure_dir_exists( dirname( $filepath ) );
-						$result = file_put_contents( $filepath, $markdown->get_contents() ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
-					}
-				} catch ( Exception $exception ) {
-					WP_CLI::error(
-						"Failed to generate markdown files: {$exception->getMessage()}\n{$exception->getTraceAsString()}"
-					);
-				}
-				break;
-			case '':
-				WP_CLI::error( "A value of 'json' or 'markdown' is required for the --format flag." );
-				break;
-			default:
-				WP_CLI::error( "Invalid --format value '{$format}' provided. Possible values: json, markdown" );
-		}
-
-		WP_CLI::line();
-
+		$output_file = $destination_folder . '/docs.json';
+		$json        = wp_json_encode( $data, JSON_PRETTY_PRINT );
+		$result      = file_put_contents( $output_file, $json ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
 		if ( false === $result ) {
 			WP_CLI::error( "Problem writing data to file '{$output_file}'" );
 			exit;
 		}
+		WP_CLI::line();
+		WP_CLI::success( "Generated JSON data saved to '{$output_file}'." );
+
+		try {
+			$doc_tree = new Root( $data );
+		} catch ( Exception $exception ) {
+			WP_CLI::error(
+				"Failed to build documentation object tree: {$exception->getMessage()}\n{$exception->getTraceAsString()}",
+				false // Using separate exit for PHPStan.
+			);
+			exit;
+		}
+
+		// Empty out all markdown files located inside of the directories.
+		$md_dirs = [
+			'class',
+			'hook',
+			'function',
+			'method',
+		];
+		foreach ( $md_dirs as $md_dir ) {
+			$iterator = new RegexIterator(
+				new RecursiveIteratorIterator(
+					new RecursiveDirectoryIterator( $destination_folder . '/' . $md_dir )
+				),
+				'/\.md$/'
+			);
+			foreach ( $iterator as $file ) {
+				if ( unlink( $file ) ) {
+					WP_CLI::line( "Cleaned: $file" );
```

Comment:
Maybe this should be behind a verbose flag? Also adding a success notice when done cleaning would be a nice addition as well.

Reply:
I've added a success message indicating the count in 713c56f3e.

However, I prefer to list out all cleaned files irrespective of a `--verbose` flag. To me it is key to get a sanity check on the files that got deleted, because if I see anything that was removed unexpectedly I want to know.

Reply:
Understood, better to be safe than sorry.